packages/wme-ui/src/components/setup-card-task/setup-card-task.tsx

```diff
@@ -2,223 +2,120 @@ import React from 'react';
+  name: 'WmeTaskIcon',
   slot: 'Root',
-})<BoxProps>(({ theme }) => ({
+})<Pick<SetupCardTaskProps, 'children'>>(({ theme }) => ({
+  position: 'relative',
   display: 'flex',
+  justifyContent: 'center',
   alignItems: 'center',
-  marginLeft: 'auto',
-  flex: '0 0 auto',
-  color: theme.palette.text.primary,
-}));
-
-const CtaAction = styled(Typography, {
-  name: 'WmeTaskCta',
-  slot: 'Action',
-})<TypographyProps>(() => ({
-  fontWeight: 600,
-  letterSpacing: '-0.25px',
-  lineHeight: pxToRem(26),
-  opacity: 0,
-  transform: 'translateX(-10px)',
-  transition: 'all 0.3s ease-in-out',
-
-  '.MuiButtonBase-root:hover &, .Mui-focusVisible &': {
-    opacity: 1,
-    transform: 'translateX(0)',
-  },
-}));
-
-const TaskActionArea = styled(CardActionArea)<TaskActionAreaProps>(({ theme }) => ({
-  '&.Mui-focusVisible': {
-    boxShadow: 'none',
-    outline: 'none',
-  },
-  '& .MuiCardActionArea-focusHighlight': {
-    backgroundColor: theme.palette.grey[600],
+  zIndex: 1,
+  marginRight: theme.spacing(2),
+  width: 60,
+  height: 60,
+  backgroundColor: theme.palette.grey[100],
+  borderRadius: '50%',
+  transition: theme?.transitions?.create(['background-color'], {
+    duration: theme.transitions.duration.standard,
+  }),
+  '& .MuiSvgIcon-root': {
+    height: '1.2rem',
+    width: '1.2rem',
   },
 }));
 
-const CtaActionRoot = (props: CtaActionRootProps) => {
-  const { taskCta } = props;
-  return (
-    <SetupCardTaskCta className="WmeTaskCta-root">
-      <CtaAction className="WmeTaskCta-action" variant="body2">
-        {taskCta}
-      </CtaAction>
-      <ChevronRight />
-    </SetupCardTaskCta>
-  );
-};
-
-const SetupCardTask: React.FC<SetupCardTaskProps> = (props) => {
+const SetupCardTask = (props: SetupCardTaskProps) => {
   const {
-    href = '',
     title,
     intro,
-    button: buttonProp,
-    avatar,
-    variant,
-    taskCta,
+    icon,
+    href,
     onClick,
-    children,
-    disabled = false,
+    button,
+    disabled,
+    taskCta,
+    variant,
+    action,
+    avatar,
   } = props;
 
-  // Check that `button` is React Element and `variant` is `action`.
-  const variantType = React.isValidElement(buttonProp) && variant === 'action' ? 'action' : 'task';
-
-  // If SetupCardTask is `disabled` add `disabled` prop to
-  let button = (buttonProp && disabled)
-    ? React.cloneElement(buttonProp as React.ReactElement<any>, { disabled: true })
-    : buttonProp;
+  const deprecatedProps = { variant, action, avatar };
 
-  const { props: buttonProps } = button || {};
-
-  // If button has no `onClick` handler, we attribute the parent `onClick` handler.
-  button = (button && (typeof buttonProps === 'object' && !('onClick' in buttonProps)))
-    ? React.cloneElement(button as React.ReactElement<any>, { onClick })
-    : button;
+  if (Object.keys(deprecatedProps).length > 0) {
+    console.error(`You are using a deprecated prop for the Setup Card Task: ${Object.keys(deprecatedProps).join(',')}`);
+  }
```

Comment:
It could be handy to create a utility function for those deprecation errors if it's going to be a thing in the future.

Reply:
Great call!